### PR TITLE
(HC-54) Implement missing fallback/merge methods

### PR DESCRIPTION
**Pull request by MikaelSmith**
A number of fallback and merge overloads were missing, causing merges to
be skipped. Implement them.

### CODE REVIEW

File: lib/tests/concatenation_test.cc [hunk 1]
```diff
@@ -12,14 +12,15 @@ TEST_CASE("concatenation") {
...
-    SECTION("trivial string concatenation") {
-        auto conf = parse_config(R"(a : ${x}foo, x = 1)")->resolve();
-        REQUIRE(conf->get_string("a") == "1foo");
-    }
-
-    SECTION("variable lookup o:") {
-        auto conf = parse_config(R"(a : ${x}, x = 1)")->resolve();
-        REQUIRE(conf->get_string("a") == "1");
-    }
+// Tests fail now because resolve_result::resolve implementation is incomplete.
```
Comment (branan) [inline lib/tests/concatenation_test.cc]: What's the failure you're seeing? I was going to dive into finishing up resolve today/tomorrow, so I kind of want to get that in so we can uncomment these

Comment (branan) [inline lib/tests/concatenation_test.cc]: I may just base my work on this PR and sort it out myself

**Comment by branan** [comment]
I think we'll probably catch up on the coverage loss as we add more tests, so I'm fine letting that slip.

File: lib/tests/concatenation_test.cc (hunk 1 above)
Comment (MikaelSmith) [inline lib/tests/concatenation_test.cc]: Hmm, I'm not sure. I may have fixed it later. I'll restore the tests.

Comment (branan) [inline lib/tests/concatenation_test.cc]: Ah, yup. That's one of the things I was going to try to tackle today/tomorrow. I guess we can just merge this with those tests disabled, and I'll uncomment them when I get that stuff fixed

Comment (branan) [inline lib/tests/concatenation_test.cc]: Does catch have a better way of flagging tests pending than commenting them out?

Comment (MikaelSmith) [inline lib/tests/concatenation_test.cc]: Yes, https://github.com/philsquared/Catch/blob/master/docs/test-cases-and-sections.md#special-tags. I'll update to use `[!shouldfail]`.

Comment (MikaelSmith) [inline lib/tests/concatenation_test.cc]: Updated, hopefully it works in CI.

**Pull request merged**